builtin/audit/socket/backend.go

```diff
@@ -88,208 +84,70 @@ func Factory(_ context.Context, conf *audit.BackendConfig, useEventLogger bool,
-
-	err := b.write(ctx, buf.Bytes())
+	err = b.configureFormatterNode(cfg, opts...)
 	if err != nil {
-		rErr := b.reconnect(ctx)
-		if rErr != nil {
-			err = multierror.Append(err, rErr)
-		} else {
-			// Try once more after reconnecting
-			err = b.write(ctx, buf.Bytes())
-		}
+		return nil, fmt.Errorf("%s: error configuring formatter node: %w", op, err)
 	}
 
-	return err
-}
-
-// Deprecated: Use eventlogger.
-func (b *Backend) LogResponse(ctx context.Context, in *logical.LogInput) error {
-	var buf bytes.Buffer
-	if err := b.formatter.FormatAndWriteResponse(ctx, &buf, in); err != nil {
-		return err
+	sinkOpts := []event.Option{
+		event.WithSocketType(socketType),
+		event.WithMaxDuration(writeDeadline),
 	}
 
-	b.Lock()
-	defer b.Unlock()
-
-	err := b.write(ctx, buf.Bytes())
+	err = b.configureSinkNode(conf.MountPath, address, cfg.RequiredFormat.String(), sinkOpts...)
 	if err != nil {
-		rErr := b.reconnect(ctx)
-		if rErr != nil {
-			err = multierror.Append(err, rErr)
-		} else {
-			// Try once more after reconnecting
-			err = b.write(ctx, buf.Bytes())
-		}
+		return nil, fmt.Errorf("%s: error configuring sink node: %w", op, err)
 	}
 
-	return err
+	return b, nil
 }
 
-func (b *Backend) LogTestMessage(ctx context.Context, in *logical.LogInput, config map[string]string) error {
-	// Event logger behavior - manually Process each node
+func (b *Backend) LogTestMessage(ctx context.Context, in *logical.LogInput) error {
 	if len(b.nodeIDList) > 0 {
 		return audit.ProcessManual(ctx, in, b.nodeIDList, b.nodeMap)
 	}
 
-	// Old behavior
-	var buf bytes.Buffer
-
-	temporaryFormatter, err := audit.NewTemporaryFormatter(config["format"], config["prefix"])
-	if err != nil {
-		return err
-	}
-
-	if err = temporaryFormatter.FormatAndWriteRequest(ctx, &buf, in); err != nil {
-		return err
-	}
-
-	b.Lock()
-	defer b.Unlock()
-
-	err = b.write(ctx, buf.Bytes())
-	if err != nil {
-		rErr := b.reconnect(ctx)
-		if rErr != nil {
-			err = multierror.Append(err, rErr)
-		} else {
-			// Try once more after reconnecting
-			err = b.write(ctx, buf.Bytes())
-		}
-	}
-
-	return err
-}
-
-// Deprecated: Use eventlogger.
-func (b *Backend) write(ctx context.Context, buf []byte) error {
-	if b.connection == nil {
-		if err := b.reconnect(ctx); err != nil {
-			return err
-		}
-	}
-
-	err := b.connection.SetWriteDeadline(time.Now().Add(b.writeDuration))
-	if err != nil {
-		return err
-	}
-
-	_, err = b.connection.Write(buf)
-	if err != nil {
-		return err
-	}
-
 	return nil
 }
 
-// Deprecated: Use eventlogger.
-func (b *Backend) reconnect(ctx context.Context) error {
-	if b.connection != nil {
-		b.connection.Close()
-		b.connection = nil
-	}
-
-	timeoutContext, cancel := context.WithTimeout(ctx, b.writeDuration)
-	defer cancel()
-
-	dialer := net.Dialer{}
-	conn, err := dialer.DialContext(timeoutContext, b.socketType, b.address)
-	if err != nil {
-		return err
+func (b *Backend) Reload(ctx context.Context) error {
+	if len(b.nodeMap) > 0 {
```

Comment:
This if statement is unnecessary, the for-range loop will gracefully handle the case where b.nodeMap is empty.